### ISSUE TITLE
Lacking values in a long xml documents

### ISSUE THREAD

**Issue by ivanceras** [sacooper/xmlJSON-rs#2]
Hi,
I was trying to convert an xml document from ODS format, but it seems the converted json format is lacking

``` xml

<?xml version="1.0" encoding="UTF-8"?>
<office:document-content
    xmlns:office="urn:oasis:names:tc:opendocument:
    xmlns:office:1.0"
    xmlns:style="urn:oasis:names:tc:opendocument:
    xmlns:style:1.0"
    xmlns:text="urn:oasis:names:tc:opendocument:
    xmlns:text:1.0"
    xmlns:table="urn:oasis:names:tc:opendocument:
    xmlns:table:1.0"
    xmlns:draw="urn:oasis:names:tc:opendocument:
    xmlns:drawing:1.0"
    xmlns:fo="urn:oasis:names:tc:opendocument:
    xmlns:xsl-fo-compatible:1.0"
    xmlns:xlink="http://www.w3.org/1999/xlink"
    xmlns:dc="http://purl.org/dc/elements/1.1/"
    xmlns:meta="urn:oasis:names:tc:opendocument:
    xmlns:meta:1.0"
    xmlns:number="urn:oasis:names:tc:opendocument:
    xmlns:datastyle:1.0"
    xmlns:presentation="urn:oasis:names:tc:opendocument:
    xmlns:presentation:1.0"
    xmlns:svg="urn:oasis:names:tc:opendocument:
    xmlns:svg-compatible:1.0"
    xmlns:chart="urn:oasis:names:tc:opendocument:
    xmlns:chart:1.0"
    xmlns:dr3d="urn:oasis:names:tc:opendocument:
    xmlns:dr3d:1.0"
    xmlns:math="http://www.w3.org/1998/Math/MathML"
    xmlns:form="urn:oasis:names:tc:opendocument:
    xmlns:form:1.0"
    xmlns:script="urn:oasis:names:tc:opendocument:
    xmlns:script:1.0"
    xmlns:ooo="http://openoffice.org/2004/office"
    xmlns:ooow="http://openoffice.org/2004/writer"
    xmlns:oooc="http://openoffice.org/2004/calc"
    xmlns:dom="http://www.w3.org/2001/xml-events"
    xmlns:xforms="http://www.w3.org/2002/xforms"
    xmlns:xsd="http://www.w3.org/2001/XMLSchema"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:rpt="http://openoffice.org/2005/report"
    xmlns:of="urn:oasis:names:tc:opendocument:
    xmlns:of:1.2"
    xmlns:xhtml="http://www.w3.org/1999/xhtml"
    xmlns:grddl="http://www.w3.org/2003/g/data-view#"
    xmlns:tableooo="http://openoffice.org/2009/table"
    xmlns:drawooo="http://openoffice.org/2010/draw"
    xmlns:calcext="urn:org:documentfoundation:names:experimental:calc:
    xmlns:calcext:1.0"
    xmlns:loext="urn:org:documentfoundation:names:experimental:office:
    xmlns:loext:1.0"
    xmlns:field="urn:openoffice:names:experimental:ooo-ms-interop:
    xmlns:field:1.0"
    xmlns:formx="urn:openoffice:names:experimental:ooxml-odf-interop:
    xmlns:form:1.0"
    xmlns:css3t="http://www.w3.org/TR/css3-text/" office:version="1.2">
    <office:scripts/>
    <office:font-face-decls>
        <style:font-face style:name="Liberation Sans" svg:font-family="&apos;Liberation Sans&apos;" style:font-family-generic="swiss" style:font-pitch="variable"/>
        <style:font-face style:name="DejaVu Sans" svg:font-family="&apos;DejaVu Sans&apos;" style:font-family-generic="system" style:font-pitch="variable"/>
        <style:font-face style:name="Droid Sans Fallback" svg:font-family="&apos;Droid Sans Fallback&apos;" style:font-family-generic="system" style:font-pitch="variable"/>
        <style:font-face style:name="FreeSans" svg:font-family="FreeSans" style:font-family-generic="system" style:font-pitch="variable"/>
    </office:font-face-decls>
    <office:automatic-styles>
        <style:style style:name="co1" style:family="table-column">
            <style:table-column-properties fo:break-before="auto" style:column-width="7.257cm"/>
        </style:style>
        <style:style style:name="co2" style:family="table-column">
            <style:table-column-properties fo:break-before="auto" style:column-width="4.392cm"/>
        </style:style>
        <style:style style:name="co3" style:family="table-column">
            <style:table-column-properties fo:break-before="auto" style:column-width="5.918cm"/>
        </style:style>
        <style:style style:name="co4" style:family="table-column">
            <style:table-column-properties fo:break-before="auto" style:column-width="2.678cm"/>
        </style:style>
        <style:style style:name="co5" style:family="table-column">
            <style:table-column-properties fo:break-before="auto" style:column-width="8.033cm"/>
        </style:style>
        <style:style style:name="co6" style:family="table-column">
            <style:table-column-properties fo:break-before="auto" style:column-width="2.258cm"/>
        </style:style>
        <style:style style:name="co7" style:family="table-column">
            <style:table-column-properties fo:break-before="auto" style:column-width="6.625cm"/>
        </style:style>
        <style:style style:name="co8" style:family="table-column">
            <style:table-column-properties fo:break-before="auto" style:column-width="6.87cm"/>
        </style:style>
        <style:style style:name="ro1" style:family="table-row">
            <style:table-row-properties style:row-height="0.452cm" fo:break-before="auto" style:use-optimal-row-height="true"/>
        </style:style>
        <style:style style:name="ro2" style:family="table-row">
            <style:table-row-properties style:row-height="1.711cm" fo:break-before="auto" style:use-optimal-row-height="true"/>
        </style:style>
        <style:style style:name="ta1" style:family="table" style:master-page-name="Default">
            <style:table-properties table:display="true" style:writing-mode="lr-tb"/>
        </style:style>
        <style:style style:name="ce1" style:family="table-cell" style:parent-style-name="Default">
            <style:text-properties fo:font-weight="bold" style:font-weight-asian="bold" style:font-weight-complex="bold"/>
        </style:style>
    </office:automatic-styles>
    <office:body>
        <office:spreadsheet>
            <table:table table:name="product" table:style-name="ta1">
                <table:table-column table:style-name="co1" table:default-cell-style-name="Default"/>
                <table:table-column table:style-name="co2" table:default-cell-style-name="Default"/>
                <table:table-column table:style-name="co3" table:default-cell-style-name="Default"/>
                <table:table-column table:style-name="co4" table:default-cell-style-name="Default"/>
                <table:table-column table:style-name="co5" table:default-cell-style-name="Default"/>
                <table:table-column table:style-name="co6" table:number-columns-repeated="1019" table:default-cell-style-name="Default"/>
                <table:table-row table:style-name="ro1">
                    <table:table-cell table:style-name="ce1" office:value-type="string" calcext:value-type="string">
                        <text:p>name</text:p>
                    </table:table-cell>
                    <table:table-cell table:style-name="ce1" office:value-type="string" calcext:value-type="string">
                        <text:p>description</text:p>
                    </table:table-cell>
                    <table:table-cell table:style-name="ce1" office:value-type="string" calcext:value-type="string">
                        <text:p>price</text:p>
                    </table:table-cell>
                    <table:table-cell table:style-name="ce1" office:value-type="string" calcext:value-type="string">
                        <text:p>owner_id</text:p>
                    </table:table-cell>
                    <table:table-cell/>
                    <table:table-cell table:style-name="ce1" table:number-columns-repeated="1019"/>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>iphone4s</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Second hand Iphone4s</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="7000" calcext:value-type="float">
                        <text:p>7000</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A3]" office:value-type="string" office:string-value="alice" calcext:value-type="string">
                        <text:p>alice</text:p>
                    </table:table-cell>
                    <table:table-cell table:number-columns-repeated="1020"/>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>iphone3GSs</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Old Iphone3GS</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="3500" calcext:value-type="float">
                        <text:p>3500</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A3]" office:value-type="string" office:string-value="alice" calcext:value-type="string">
                        <text:p>alice</text:p>
                    </table:table-cell>
                    <table:table-cell table:number-columns-repeated="1020"/>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>ps2</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Second hand Sony Playstation Box</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="5000" calcext:value-type="float">
                        <text:p>5000</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A3]" office:value-type="string" office:string-value="alice" calcext:value-type="string">
                        <text:p>alice</text:p>
                    </table:table-cell>
                    <table:table-cell table:number-columns-repeated="1020"/>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>xbox360</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Second hand Xbob360</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="10000" calcext:value-type="float">
                        <text:p>10000</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A4]" office:value-type="string" office:string-value="bob" calcext:value-type="string">
                        <text:p>bob</text:p>
                    </table:table-cell>
                    <table:table-cell table:number-columns-repeated="1020"/>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>GoPro Hero 3+</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Slightly used GoPro Hero 3+</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="7000" calcext:value-type="float">
                        <text:p>7000</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A3]" office:value-type="string" office:string-value="alice" calcext:value-type="string">
                        <text:p>alice</text:p>
                    </table:table-cell>
                    <table:table-cell table:number-columns-repeated="1020"/>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Electric Guitar</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Generic Electric Guitar</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="8000" calcext:value-type="float">
                        <text:p>8000</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A4]" office:value-type="string" office:string-value="bob" calcext:value-type="string">
                        <text:p>bob</text:p>
                    </table:table-cell>
                    <table:table-cell table:number-columns-repeated="1020"/>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>GTX660 Ti videocard</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>2nd Nvidia Video card</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="11000" calcext:value-type="float">
                        <text:p>11000</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A5]" office:value-type="string" office:string-value="mary" calcext:value-type="string">
                        <text:p>mary</text:p>
                    </table:table-cell>
                    <table:table-cell table:number-columns-repeated="1020"/>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Stationary Bike</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Time Sports Stationary Bike</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="3000" calcext:value-type="float">
                        <text:p>3000</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A3]" office:value-type="string" office:string-value="alice" calcext:value-type="string">
                        <text:p>alice</text:p>
                    </table:table-cell>
                    <table:table-cell table:number-columns-repeated="1020"/>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>HP printer</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Second HP printer</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="1000" calcext:value-type="float">
                        <text:p>1000</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A3]" office:value-type="string" office:string-value="alice" calcext:value-type="string">
                        <text:p>alice</text:p>
                    </table:table-cell>
                    <table:table-cell table:number-columns-repeated="1020"/>
                </table:table-row>
                <table:table-row table:style-name="ro2">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Skyworth 42&quot; LCD TV - </text:p>
                        <text:p>This has never been used</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>{&quot;specs&quot;:&quot;Wide screen TV&quot;,</text:p>
                        <text:p> &quot;width&quot; : 1920,</text:p>
                        <text:p>&quot;height&quot; : 1080,</text:p>
                        <text:p>&quot;resolution&quot;: &quot;1080p&quot;}</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="10000" calcext:value-type="float">
                        <text:p>10000</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A3]" office:value-type="string" office:string-value="alice" calcext:value-type="string">
                        <text:p>alice</text:p>
                    </table:table-cell>
                    <table:table-cell table:number-columns-repeated="1020"/>
                </table:table-row>
            </table:table>
            <table:table table:name="users" table:style-name="ta1">
                <table:table-column table:style-name="co7" table:number-columns-repeated="2" table:default-cell-style-name="Default"/>
                <table:table-column table:style-name="co6" table:default-cell-style-name="Default"/>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>username</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>email</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>name</text:p>
                    </table:table-cell>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>SuperUser</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>bazaar@ivanceras.com</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Super Users</text:p>
                    </table:table-cell>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>alice</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>alice@acme.com</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Alice Smith</text:p>
                    </table:table-cell>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>bob</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>bob@acme.com</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Bob Pearson</text:p>
                    </table:table-cell>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>mary</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>mary@gmail.com</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>Mary Winsteaud</text:p>
                    </table:table-cell>
                </table:table-row>
            </table:table>
            <table:table table:name="review" table:style-name="ta1">
                <table:table-column table:style-name="co8" table:default-cell-style-name="Default"/>
                <table:table-column table:style-name="co6" table:default-cell-style-name="Default"/>
                <table:table-column table:style-name="co8" table:default-cell-style-name="Default"/>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>comment</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>rating</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>user_id</text:p>
                    </table:table-cell>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>This product is good, I&apos;m giving it with 4/5</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="4" calcext:value-type="float">
                        <text:p>4</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[users.A3]" office:value-type="string" office:string-value="alice" calcext:value-type="string">
                        <text:p>alice</text:p>
                    </table:table-cell>
                </table:table-row>
                <table:table-row table:style-name="ro1">
                    <table:table-cell office:value-type="string" calcext:value-type="string">
                        <text:p>This seller really sells good products</text:p>
                    </table:table-cell>
                    <table:table-cell office:value-type="float" office:value="5" calcext:value-type="float">
                        <text:p>5</text:p>
                    </table:table-cell>
                    <table:table-cell table:formula="of:=[product.A2]" office:value-type="string" office:string-value="iphone4s" calcext:value-type="string">
                        <text:p>iphone4s</text:p>
                    </table:table-cell>
                </table:table-row>
            </table:table>
            <table:named-expressions/>
        </office:spreadsheet>
    </office:body>
</office:document-content>

```

The converted json is only

``` js

Object(
    {
        "document-content": Object(
            {
                "$": Object(
                    {
                        "version": String(
                            "1.2"
                        )
                    }
                ),
                "automatic-styles": Object(
                    {
                        "style": Object(
                            {
                                "$": Object(
                                    {
                                        "family": String(
                                            "table-cell"
                                        ),
                                        "name": String(
                                            "ce1"
                                        ),
                                        "parent-style-name": String(
                                            "Default"
                                        )
                                    }
                                ),
                                "text-properties": Object(
                                    {
                                        "$": Object(
                                            {
                                                "font-weight": String(
                                                    "bold"
                                                ),
                                                "font-weight-asian": String(
                                                    "bold"
                                                ),
                                                "font-weight-complex": String(
                                                    "bold"
                                                )
                                            }
                                        )
                                    }
                                )
                            }
                        )
                    }
                ),
                "body": Object(
                    {
                        "spreadsheet": Object(
                            {
                                "named-expressions": Object(
                                    {}
                                ),
                                "table": Object(
                                    {
                                        "$": Object(
                                            {
                                                "name": String(
                                                    "review"
                                                ),
                                                "style-name": String(
                                                    "ta1"
                                                )
                                            }
                                        ),
                                        "table-column": Object(
                                            {
                                                "$": Object(
                                                    {
                                                        "default-cell-style-name": String(
                                                            "Default"
                                                        ),
                                                        "style-name": String(
                                                            "co8"
                                                        )
                                                    }
                                                )
                                            }
                                        ),
                                        "table-row": Object(
                                            {
                                                "$": Object(
                                                    {
                                                        "style-name": String(
                                                            "ro1"
                                                        )
                                                    }
                                                ),
                                                "table-cell": Object(
                                                    {
                                                        "$": Object(
                                                            {
                                                                "formula": String(
                                                                    "of:=[product.A2]"
                                                                ),
                                                                "string-value": String(
                                                                    "iphone4s"
                                                                ),
                                                                "value-type": String(
                                                                    "string"
                                                                )
                                                            }
                                                        ),
                                                        "p": Object(
                                                            {
                                                                "_": String(
                                                                    "iphone4s"
                                                                )
                                                            }
                                                        )
                                                    }
                                                )
                                            }
                                        )
                                    }
                                )
                            }
                        )
                    }
                ),
                "font-face-decls": Object(
                    {
                        "font-face": Object(
                            {
                                "$": Object(
                                    {
                                        "font-family": String(
                                            "FreeSans"
                                        ),
                                        "font-family-generic": String(
                                            "system"
                                        ),
                                        "font-pitch": String(
                                            "variable"
                                        ),
                                        "name": String(
                                            "FreeSans"
                                        )
                                    }
                                )
                            }
                        )
                    }
                ),
                "scripts": Object(
                    {}
                )
            }
        )
    }
)


```

**Comment by sacooper** [comment]
Thanks for bringing this to my attention. I've taken a look, and I have a rough idea of what's going on. I'll try to work on this in the near future.

**Comment by sacooper** [comment]
I was able to fix the missing elements, but there also seem to be some issues with namespaces. Once I've fixed that I'll publish a new version

**Comment by sacooper** [comment]
I've fixed the main issue with missing values here: 02dac79619d146c1d9e0226475faf8216c009e83 but there's still an issue with namespaces I need to fix. You can always reference this commit from your Cargo.toml, and update to 0.2.0 when I fix the namespace issue

**Comment by ivanceras** [comment]
Thank you for your updates. I appreciated it.

**Comment by sacooper** [comment]
Once again, really sorry for the delay in pushing out updates. I've just published v0.2.0, and this issue, along with a few others (like returning an Err instead of panicking on invalid XML) have been fixed. 

Thanks for your help with posting the xml. It played a huge role in helping me tests this.

If you find any other issues, or have any features you think would make this more usable, feel free to open another issue!
